### PR TITLE
[SMTChecker] Fix undefined behavior with nullptr

### LINKED ISSUE #8674: [SMTChecker] Runtime error: reference binding to to null pointer of type `const char*`
## Description

```
pragma experimental SMTChecker;
contract C
{
        uint a;
        function f(uint x) public { f(0); f; }
        function g() public { f(1); f(0); }
}
```

throws the runtime error here

https://github.com/ethereum/solidity/blob/95349b36349818288638521e4156cd1a40d6be77/libsolidity/formal/Z3Interface.cpp#L240

probably because the referenced vector member may have been uninitialized because an earlier for loop that was supposed to initialize the said vector was not taken for some reason

https://github.com/ethereum/solidity/blob/95349b36349818288638521e4156cd1a40d6be77/libsolidity/formal/Z3Interface.cpp#L229-L230

## Environment

- Compiler version: latest develop

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes https://github.com/ethereum/solidity/issues/8674

### CODE REVIEW

File: libsolidity/formal/Z3Interface.cpp [hunk 1]
```diff
@@ -237,8 +237,8 @@ z3::sort Z3Interface::z3Sort(Sort const& _sort)
 		z3::func_decl tupleConstructor = m_context.tuple_sort(
 			tupleSort.name.c_str(),
 			tupleSort.members.size(),
-			&cMembers[0],
-			&sorts[0],
+			cMembers.empty() ? nullptr : &cMembers[0],
+			sorts.empty() ? nullptr : &sorts[0],
```
Comment (axic) [inline libsolidity/formal/Z3Interface.cpp]: `cMembers.data()` ?

See https://www.cplusplus.com/reference/vector/vector/data/

Comment (leonardoalt) [reply libsolidity/formal/Z3Interface.cpp]: Ah nice

Comment (leonardoalt) [reply libsolidity/formal/Z3Interface.cpp]: Fixed

**Review by axic — APPROVED** [review]

**Comment by bshastry** [comment]
Can confirm that this PR fixes the tracked issue.

**Pull request merged**